### PR TITLE
expression: handle max_allowed_packet warnings for pad functions

### PR DESCRIPTION
**Pull request by zz-jason**
## What have you changed? (mandatory)

Return `NULL` and a warning when the result of `lpad` or `rpad` exceeds `max_allowed_packetbs`.

Before this PR, `select rpad('hello', 18446744073709551617, '1');` would make tidb-server panic by `runtime error: makeslice: len out of range`.

After this PR:
```sql
TiDB(localhost:4000) > select rpad('hello', 18446744073709551617, '1');
+------------------------------------------+
| rpad('hello', 18446744073709551617, '1') |
+------------------------------------------+
| NULL                                     |
+------------------------------------------+
1 row in set, 2 warnings (0.00 sec)

TiDB(localhost:4000) > show warnings;
+---------+------+----------------------------------------------------------------------------+
| Level   | Code | Message                                                                    |
+---------+------+----------------------------------------------------------------------------+
| Warning | 1292 | Truncated incorrect DECIMAL value: '18446744073709551617'                  |
| Warning | 1301 | Result of rpad() was larger than max_allowed_packet (67108864) - truncated |
+---------+------+----------------------------------------------------------------------------+
2 rows in set (0.00 sec)
```

## What is the type of the changes? (mandatory)

- Bug fix (non-breaking change which fixes an issue)

## How has this PR been tested? (mandatory)

- unit test
- explain test

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

No

## Does this PR need to be added to the release notes? (mandatory)

Yes

release note:
```
Return `NULL` when the result of function `LPAD`/`RPAD` is larger than `max_allowed_packet`
```

## Refer to a related PR or issue link (optional)

to #7153

### CODE REVIEW

**Comment by zz-jason** [comment]
@winoros @XuHuaiyu PTAL

File: expression/builtin_string.go [hunk 1]
```diff
@@ -1760,24 +1760,33 @@ func (c *lpadFunctionClass) getFunction(ctx sessionctx.Context, args []Expressio
 	bf.tp.Flen = getFlen4LpadAndRpad(bf.ctx, args[1])
 	SetBinFlagOrBinStr(args[0].GetType(), bf.tp)
 	SetBinFlagOrBinStr(args[2].GetType(), bf.tp)
+
+	valStr, _ := ctx.GetSessionVars().GetSystemVar("max_allowed_packet")
```
Comment (XuHuaiyu) [inline expression/builtin_string.go]: make "max_allowed_packet" a constant in sessionctx/variable/sysvar.go?

File: mysql/errname.go [hunk 2]
```diff
@@ -316,7 +316,7 @@ var MySQLErrName = map[uint16]string{
 	ErrUnknownTimeZone:                          "Unknown or incorrect time zone: '%-.64s'",
 	ErrWarnInvalidTimestamp:                     "Invalid TIMESTAMP value in column '%s' at row %d",
 	ErrInvalidCharacterString:                   "Invalid %s character string: '%.64s'",
-	ErrWarnAllowedPacketOverflowed:              "Result of %s() was larger than maxAllowedPacket (%d) - truncated",
+	ErrWarnAllowedPacketOverflowed:              "Result of %s() was larger than max_allowed_packet (%d) - truncated",
```
Comment (winoros) [inline mysql/errname.go]: This is original warning msg in mysql?

Comment (zz-jason) [reply mysql/errname.go]: Yes, this error message comes from the mysql-server code base.

**Comment by zz-jason** [comment]
@XuHuaiyu Done, PTAL

**Pull request merged**